apps/api/src/goods-receipts/goods-receipts.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  UsePipes,
+  ValidationPipe,
+  BadRequestException
+} from '@nestjs/common';
+import { GoodsReceiptsService } from './goods-receipts.service';
+import { CreateGoodsReceiptDto } from './dto/create-goods-receipt.dto';
+import { UpdateGoodsReceiptDto } from './dto/update-goods-receipt.dto';
+
+@Controller('goods-receipts')
+export class GoodsReceiptsController {
+  constructor(private readonly goodsReceiptsService: GoodsReceiptsService) {}
+
+  @Post()
+  @UsePipes(new ValidationPipe({ transform: true }))
+  create(@Body() createGoodsReceiptDto: CreateGoodsReceiptDto) {
+    return this.goodsReceiptsService.create(createGoodsReceiptDto);
+  }
+
+  @Get()
+  findAll() {
+    return this.goodsReceiptsService.findAll();
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    const numericId = Number(id);
+    if (Number.isNaN(numericId)) {
+      throw new BadRequestException('Invalid ID format');
+    }
+    return this.goodsReceiptsService.findOne(numericId);
+  }
+
+  @Patch(':id')
+  @UsePipes(new ValidationPipe({ transform: true }))
+  update(
+    @Param('id') id: string,
+    @Body() updateGoodsReceiptDto: UpdateGoodsReceiptDto
+  ) {
+    return this.goodsReceiptsService.update(+id, updateGoodsReceiptDto);
```

Comment:
Do you think +id is a bit of a cryptic way to transform something? With class transformer setup, you should be able to make the id type a number and have it come in that way.

Reply:
I do think that +id is cryptic. This was generated by the Next.js CRUD generator and I refactored in some places to cast it to a number rather than using the + operator but missed updating it everywhere. 

I didn't think about creating a UpdateGoodsReceiptParam object for the parameters and applying the transform to id similar to how I handled UpdateGoodsReceiptDto. Thanks for the feedback!